lib/trino-parquet/src/test/java/io/trino/parquet/crypto/TestParquetEncryption.java

```diff
@@ -0,0 +1,1057 @@
+import org.apache.parquet.hadoop.example.ExampleParquetWriter;
+import org.apache.parquet.hadoop.metadata.ColumnPath;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.MessageTypeParser;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Types;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import static io.trino.memory.context.AggregatedMemoryContext.newSimpleAggregatedMemoryContext;
+import static io.trino.parquet.predicate.PredicateUtils.getFilteredRowGroups;
+import static io.trino.spi.predicate.Domain.singleValue;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.parquet.hadoop.ParquetFileWriter.Mode.OVERWRITE;
+import static org.apache.parquet.hadoop.metadata.CompressionCodecName.SNAPPY;
+import static org.apache.parquet.hadoop.metadata.CompressionCodecName.UNCOMPRESSED;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.joda.time.DateTimeZone.UTC;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+/**
+ * Unit‑tests exercising Trino’s PME path.
+ */
+// ExampleParquetWriter is not thread-safe
+@TestInstance(PER_CLASS)
+@Execution(ExecutionMode.SAME_THREAD)
+public class TestParquetEncryption
+{
+    // ──────────────────────────────────  CONSTANTS  ────────────────────────────────────
+    private static final ColumnPath AGE_PATH = ColumnPath.fromDotString("age");
+    private static final ColumnPath ID_PATH = ColumnPath.fromDotString("id");
+
+    private static final byte[] KEY_AGE = "colKeyIs16ByteA?".getBytes(UTF_8);
+    private static final byte[] KEY_ID = "colKeyIs16ByteB?".getBytes(UTF_8);
+    private static final byte[] KEY_FOOT = "footKeyIs16Byte?".getBytes(UTF_8);
+
+    // one‑column schema
+    private static final MessageType AGE_SCHEMA =
+            MessageTypeParser.parseMessageType("message doc { required int32 age; }");
+
+    // two‑column schema
+    private static final MessageType TWO_COL_SCHEMA = MessageTypeParser.parseMessageType(
+            "message doc { required int32 age; required int32 id; }");
+
+    // ────────────────────────────────  PARAMETERISED TESTS  ─────────────────────────────
+
+    /**
+     * Column encryption only – footer left in plaintext.
+     */
+    @ParameterizedTest(name = "columnOnly_footerIntegrity={0}_compressed={1}")
```

Comment:
JFYI test parameterization is discouraged https://trino.io/docs/current/develop/tests.html#avoid-data-providers-and-parametric-tests
But I think parameterization is more readable and debuggable